File: robotiq_2f_gripper_control/src/robotiq_2f_gripper_control/gripper_action_client.py
#!/usr/bin/env python3

import rospy
import actionlib
import robotiq_2f_gripper_control.msg

def feedback_cb(feedback):
    rospy.loginfo("Feedback: Current Position - %i" % feedback.current_position)

def done_cb(status, result):
    if status == actionlib.GoalStatus.SUCCEEDED:
        rospy.loginfo("Action Succeeded: Result - %s" % result.result)
    else:
        rospy.loginfo("Action Failed with status: %d" % status)

def main():
    rospy.init_node('gripper_action_client')

    # Create an action client
    client = actionlib.SimpleActionClient('gripper_action_server', robotiq_2f_gripper_control.msg.gripAction)

    # Wait for the action server to start
    rospy.loginfo("Waiting for action server to start...")
    client.wait_for_server()
    rospy.loginfo("Action server started, sending goal...")

    # Create a goal to send
    goal = robotiq_2f_gripper_control.msg.gripGoal()
    goal.position = 100  # Desired position
    goal.speed = 50      # Speed
    goal.force = 20      # Force

    # Send the goal and register callbacks
    client.send_goal(goal)

    # Wait for the result
    client.wait_for_result()

    rospy.loginfo("Result: %s" % client.get_result().result)

if __name__ == '__main__':
    main()

File: robotiq_2f_gripper_control/src/gripper_action_client.cpp
#include "ros/console.h"
#include <ros/ros.h>
#include <actionlib/client/simple_action_client.h>
#include <robotiq_2f_gripper_control/gripAction.h>

// Define a type for the action client
typedef actionlib::SimpleActionClient<robotiq_2f_gripper_control::gripAction> GripperClient;


int main(int argc, char** argv) {
    // Initialize the ROS node
    ros::init(argc, argv, "gripper_action_client");
    ros::NodeHandle nh("~");

    int goal_position = 0;
    nh.getParam("goal_position", goal_position);
    ROS_INFO("Goal position : %d", goal_position);

    // Create an action client
    GripperClient client("gripper_action_server", true);

    // Wait for the action server to start
    ROS_INFO("Waiting for action server to start...");
    client.waitForServer();
    ROS_INFO("Action server started, sending goal...");

    // Create a goal to send
    robotiq_2f_gripper_control::gripGoal goal;
    goal.position = goal_position;  // Desired position
    goal.speed = 50;      // Speed
    goal.force = 20;      // Force

    // Send the goal and register callbacks
    client.sendGoal(goal);

    // Wait for the result
    client.waitForResult();

    // If you want to explicitly check the result after waiting, you can do so here
    ROS_INFO("Result: %d", client.getResult()->result);

    return 0;
}
